server/src/main/java/org/opensearch/snapshots/SnapshotShardsService.java

```diff
@@ -275,42 +277,59 @@ private void startNewShards(SnapshotsInProgress.Entry entry, Map<ShardId, IndexS
                 final IndexShardSnapshotStatus snapshotStatus = shardEntry.getValue();
                 final IndexId indexId = indicesMap.get(shardId.getIndexName());
                 assert indexId != null;
-                snapshot(shardId, snapshot, indexId, entry.userMetadata(), snapshotStatus, entry.version(), new ActionListener<String>() {
-                    @Override
-                    public void onResponse(String newGeneration) {
-                        assert newGeneration != null;
-                        assert newGeneration.equals(snapshotStatus.generation());
-                        if (logger.isDebugEnabled()) {
-                            final IndexShardSnapshotStatus.Copy lastSnapshotStatus = snapshotStatus.asCopy();
-                            logger.debug(
-                                "snapshot [{}] completed to [{}] with [{}] at generation [{}]",
-                                snapshot,
-                                snapshot.getRepository(),
-                                lastSnapshotStatus,
-                                snapshotStatus.generation()
-                            );
+                if (isRemoteSnapshot(shardId)) {
+                    // If the source of the data is another remote snapshot (i.e. searchable snapshot)
+                    // then no need to snapshot the shard and can immediately notify success.
+                    notifySuccessfulSnapshotShard(snapshot, shardId, UUIDs.randomBase64UUID());
```

Comment:
Shouldn't we return the consistent generation here? `snapshotStatus.generation()` may be? or snapshot's id (it has uuid part)?